src/ui/DynamicHierarchicalFacetSearch/DynamicHierarchicalFacetSearchValueRenderer.ts

```diff
@@ -0,0 +1,73 @@
+import { IDynamicHierarchicalFacet } from '../DynamicHierarchicalFacet/IDynamicHierarchicalFacet';
+import { DynamicHierarchicalFacetSearchValue } from './DynamicHierarchicalFacetSearchValue';
+import { $$ } from '../../utils/Dom';
+
+const PATH_START_LENGTH = 1;
+const PATH_END_LENGTH = 2;
+const PATH_SEPARATOR = '/';
+
+export class DynamicHierarchicalFacetSearchValueRenderer {
+  constructor(private facetValue: DynamicHierarchicalFacetSearchValue, private facet: IDynamicHierarchicalFacet) {}
+
+  private get pathToRender() {
+    const fullPath = this.facetValue.fullPath;
+    if (fullPath.length > PATH_START_LENGTH + PATH_END_LENGTH) {
+      return [...fullPath.slice(0, PATH_START_LENGTH), '...', ...fullPath.slice(-PATH_END_LENGTH)];
+    }
+    return fullPath;
+  }
+
+  public render() {
+    const element = $$('div', { className: 'coveo-dynamic-hierarchical-facet-search-value' }, this.renderHeader(), this.renderPath());
+
+    return element.el;
+  }
+
+  public selectAction() {
+    this.facet.selectPath(this.facetValue.fullPath);
+    this.facet.enableFreezeFacetOrderFlag();
+    this.facet.enablePreventAutoSelectionFlag();
+    this.facet.scrollToTop();
+    this.facet.triggerNewQuery(() => this.facetValue.logSelectActionToAnalytics());
+  }
+
+  private renderHeader() {
+    return $$(
+      'span',
+      { className: 'coveo-dynamic-hierarchical-facet-search-value-header', ariaHidden: true },
```

Comment:
Accessibility will be improved here.

Reply:
Speaking of accessibility, I'm seeing `span`  for labels, for example. Possible to just use better semantic elements ?

Reply:
Sure, I'll change them to semantic elements.